### PR TITLE
fix mcu issue in qdb and show flags with color BLUE in uppercase, oth…

### LINKED ISSUE #1262: Qdb not working for Cortex-M mcu emulation due to missing CPSR register
**Describe the bug**
Qdb requires the CPSR register for [context rendering](https://github.com/qilingframework/qiling/blob/master/qiling/debugger/qdb/render/render_arm.py#L22) and [branch prediction](https://github.com/qilingframework/qiling/blob/master/qiling/debugger/qdb/branch_predictor/branch_predictor_arm.py). However, `reg_map` in [cortex_m_const.py](https://github.com/qilingframework/qiling/blob/master/qiling/arch/cortex_m_const.py#L9) doesn't contain a mapping for the CPSR register which leads to an AttributeError.

**Sample Code (tested with current dev branch)**
```python
from qiling.core import Qiling
from qiling.const import QL_VERBOSE
from qiling.extensions.mcu.stm32f4 import stm32f411


def test_mcu_gpio_stm32f411():
    ql = Qiling(["hello_gpioA.hex"], archtype="cortex_m", ostype="mcu", env=stm32f411, verbose=QL_VERBOSE.OFF)

    ql.hw.create("usart2").watch()
    ql.hw.create("rcc").watch()
    ql.hw.create("gpioa").watch()

    ql.hw.gpioa.hook_set(5, lambda: print("LED light up"))
    ql.hw.gpioa.hook_reset(5, lambda: print("LED light off"))

    ql.debugger = "qdb"

    ql.run(count=10000)


if __name__ == "__main__":
    test_mcu_gpio_stm32f411()
```
Results in:
```
Traceback (most recent call last):
  File "/mnt/data/Temp/a/emulation/stm32f411_gpio_hook.py", line 22, in <module>
    test_mcu_gpio_stm32f411()
  File "/mnt/data/Temp/a/emulation/stm32f411_gpio_hook.py", line 18, in test_mcu_gpio_stm32f411
    ql.run(count=10000)
  File "/mnt/data/Temp/a/emulation/qiling/core.py", line 575, in run
    debugger = debugger(self)
  File "/mnt/data/Temp/a/emulation/qiling/debugger/qdb/qdb.py", line 46, in __init__
    self.dbg_hook(init_hook)
  File "/mnt/data/Temp/a/emulation/qiling/debugger/qdb/qdb.py", line 93, in dbg_hook
    self.do_context()
  File "/mnt/data/Temp/a/emulation/qiling/debugger/qdb/qdb.py", line 384, in do_context
    self.render.context_reg(self._saved_reg_dump)
  File "/mnt/data/Temp/a/emulation/qiling/debugger/qdb/render/render.py", line 44, in wrapper
    context_dumper(*args, **kwargs)
  File "/mnt/data/Temp/a/emulation/qiling/debugger/qdb/render/render_arm.py", line 34, in context_reg
    self.print_mode_info(self.ql.arch.regs.cpsr)
  File "/mnt/data/Temp/a/emulation/qiling/arch/register.py", line 42, in __getattr__
    return super().__getattribute__(name)
AttributeError: 'QlRegisterManager' object has no attribute 'cpsr'
```

**Expected behavior**
Qdb should be executed without an AttributeError.

**Additional context**
A dirty solution is to add `"cpsr": UC_ARM_REG_CPSR,` to `reg_map` in [cortex_m_const.py](https://github.com/qilingframework/qiling/blob/master/qiling/arch/cortex_m_const.py#L9). Although this allows Qdb to run, the solution is messy because the CPSR register is not present in ARMv6-M and ARMv7-M processors ([source](https://developer.arm.com/documentation/dui0473/m/overview-of-the-arm-architecture/current-program-status-register)). My suggested solution would be to extend ARM debugging by differentiating between different ARM architecture profiles (Cortex-M vs Cortex-A vs Cortex-R). Profiles which do not implement the CPSR register need to be treated differently.

### PR DESCRIPTION
**Pull request by ucgJhe**
1. fix #1262 
2. more intuitive way for displaying flags

image

### CODE REVIEW

File: qiling/arch/cortex_m_const.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "apsr":      UC_ARM_REG_APSR,
     "ipsr":      UC_ARM_REG_IPSR,
     "epsr":      UC_ARM_REG_EPSR,
+    "cpsr":      UC_ARM_REG_CPSR,
```
Comment (elicn) [inline qiling/arch/cortex_m_const.py:34]: That is not a good idea... If you decide to go with a temporary solution, then we should do this in a different way. I suggest to do that by modifying render_arm.py line 34:
```python
# there is no cpsr in cortex m
if 'cpsr' in self.ql.arch.regs.register_mapping:
    self.print_mode_info(self.ql.arch.regs.cpsr)
```

**Review by elicn — CHANGES_REQUESTED** [review]

**Comment by xwings** [comment]
@ucgJhe will you be able to update the code ?

**Comment by ucgJhe** [comment]
@xwings will still break something, i think this can wait until we fix #1265

**Comment by xwings** [comment]
We need @elicn help for that,

**Comment by xwings** [comment]
@ucgJhe i merge Eli's PR. Will you be able to try ?

**Comment by ucgJhe** [comment]
looks good now

**Comment by ucgJhe** [comment]
new question, Im not sure is this normal ? stack pointer will point to a non-existence value

image

**Comment by elicn** [comment]
Hey @ucgJhe , what keeps us from merging this one?
I think once the `cpsr` addition to CORTEX_M is reverted, we're good to go.

**Comment by ucgJhe** [comment]
> Hey @ucgJhe , what keeps us from merging this one? I think once the `cpsr` addition to CORTEX_M is reverted, we're good to go.

It's all good to go. ping @xwings

**Review by kabeor — APPROVED** [review]

**Review by elicn — APPROVED** [review]

**Pull request merged**